pkg/gofr/datasource/opentsdb/query.go

```diff
@@ -0,0 +1,414 @@
+package opentsdb
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sort"
+	"strconv"
+	"time"
+
+	"go.opentelemetry.io/otel/trace"
+)
+
+// QueryParam is the structure used to hold the querying parameters when calling /api/query.
+// Each attributes in QueryParam matches the definition in
+// [OpenTSDB Official Docs]: http://opentsdb.net/docs/build/html/api_http/query/index.html.
+type QueryParam struct {
+	// The start time for the query. This can be a relative or absolute timestamp.
+	// The data type can only be string, int, or int64.
+	// The value is required with non-zero value of the target type.
+	Start interface{} `json:"start"`
+
+	// An end time for the query. If not supplied, the TSD will assume the local
+	// system time on the server. This may be a relative or absolute timestamp.
+	// The data type can only be string, or int64.
+	// The value is optional.
+	End interface{} `json:"end,omitempty"`
```

Comment:
```suggestion
	Start any `json:"start"`

	// An end time for the query. If not supplied, the TSD will assume the local
	// system time on the server. This may be a relative or absolute timestamp.
	// The data type can only be string, or int64.
	// The value is optional.
	End any `json:"end,omitempty"`
```